doc/programming/parsing.rst

```diff
@@ -328,11 +328,15 @@ prefix corresponding fields with:
     :show-with: foo.spicy
 
 ``skip`` works for all kinds of fields but is particularly efficient
-with ``bytes`` fields, for which it will generate optimized code
+for fields of known size for which optimized code will be generating
 avoiding the overhead of storing any data.
 
-``skip`` fields may have conditions and hooks attached, like any other
-fields. However, they do not support ``$$`` in expressions and hook.
+``skip`` fields may have conditions and hooks attached, like any other fields.
+However, they do not support ``$$`` in expressions and hook. Since ``skip``
+fields might be completely elided there is no guarantee that any stateful side
+effects from parsing them happens, e.g. ``&requires`` attributes might be
+ignored, side effects of ``&convert`` attributes might not manifest, and hooks
+could not end up being invoked.
```

Comment:
I'd phrase this more directly as "undefined", like:

```suggestion
However, they do not support ``$$`` in expressions and hook. 

Since ``skip``
allows the compiler to optimize the field's parsing code---including completely 
eliding most of it---it remains undefined if any side effects associated with the 
field will take effect. For example, ``&requires`` attributes might be ignored, 
``&convert`` expressions might not be evaluated, and hooks could end up not 
being invoked.
```